### PR TITLE
Kdb showmeta

### PR DESCRIPTION
**Pull request by Piankero**
## Basics

Check relevant points but **please do not remove entries**.
Do not describe the purpose of this PR in the PR description but:

- [ ] Short descriptions should be in the release notes (added as entry in
      `doc/news/_preparation_next_release.md` which contains `_(my name)_`)
      **Please always add something to the the release notes.**
- [ ] Longer descriptions should be in documentation or in design decisions.
- [ ] Describe details of how you changed the code in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, should be in the commit messages.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise and good to read
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the build server is happy** and also you
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by Piankero** [comment]
@markus2330 I am a bit confused on where to add tests. Should I make a shell test under `tests/shell` or a `shellrecorder` test somewhere?

**Comment by sanssecours** [comment]
> @markus2330 I am a bit confused on where to add tests. Should I make a shell test under `tests/shell` or a `shellrecorder` test somewhere?

Obviously I am not Markus, but I think using the same approach as we do for other commands like `kdb ls` should be fine. Just add [some examples invocations of the command](https://github.com/ElektraInitiative/libelektra/blob/master/doc/help/kdb-ls.md#examples) in the ReadMe using [Markdown Shell Recorder syntax](https://github.com/ElektraInitiative/libelektra/blob/master/tests/shell/shell_recorder/tutorial_wrapper/README.md), and then add the appropriate CMake code for the file somewhere here:


https://github.com/ElektraInitiative/libelektra/blob/5519cb8066a096215a3701ca3d8c02fcebe54914/tests/shell/shell_recorder/tutorial_wrapper/CMakeLists.txt#L3-L11

.

**Comment by sanssecours** [comment]
Looks like installing fails, if we do not add the (generated) man page for a certain tool. I just pushed a commit that should fix the problem.

**Comment by markus2330** [comment]
@sanssecours why do you want to not fail in that case? Imho it is good if the build server helps us when we forgot to add a man page.

**Comment by sanssecours** [comment]
> @sanssecours why do you want to not fail in that case?

We can also fail, if you want to. However, then we should provide a good error message. The current output:

```
CMake Error at doc/help/cmake_install.cmake:221 (file):
  file INSTALL cannot find
  "/home/travis/build/ElektraInitiative/libelektra/doc/man/man1/kdb-showmeta.1".
Call Stack (most recent call first):
  doc/cmake_install.cmake:74 (include)
  cmake_install.cmake:47 (include)
```

is not really helpful.

File: scripts/cmake/Modules/LibAddMacros.cmake [hunk 1]
```diff
@@ -572,7 +572,8 @@ function (generate_manpage NAME)
 			add_dependencies (man man-${NAME})
 		endif (RONN_LOC)
 
-		if (INSTALL_DOCUMENTATION)
+		# It is possible that the man page does not exist, if nobody generated it using `ronn`.
+		if (INSTALL_DOCUMENTATION AND EXISTS "${OUTFILE}")
```
Comment (markus2330) [inline scripts/cmake/Modules/LibAddMacros.cmake]: As discussed: please remove this change but add the rendered man page.

Comment (sanssecours) [reply scripts/cmake/Modules/LibAddMacros.cmake]: I removed the commit, and added the man page in commit b948543.

**Comment by markus2330** [comment]
Thank you, good job!

**Pull request merged**